packages/ethereum-contracts/contracts/superfluid/SuperToken.sol

```diff
@@ -223,6 +234,101 @@ contract SuperToken is
         return _STANDARD_DECIMALS;
     }
 
+    /**************************************************************************
+     * ERC20 Permit (EIP-2612)
+     *************************************************************************/
+
+    /// @dev EIP-2612 Permit
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public virtual override {
+        if (block.timestamp > deadline) revert SUPER_TOKEN_PERMIT_EXPIRED_SIGNATURE(deadline);
+
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _PERMIT_TYPEHASH,
+                owner,
+                spender,
+                value,
+                _nonces[owner]++,
+                deadline
+            )
+        );
+
+        bytes32 domainSeparator = DOMAIN_SEPARATOR();
```

Comment:
do not duplicate things from https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v5.3/contracts/utils/cryptography/MessageHashUtils.sol